### PR TITLE
User Edit Page

### LINKED ISSUE #65: User Story 18, User Can Edit their Profile Data
```
As a registered user
When I visit my profile page
I see a link to edit my profile data
When I click on the link to edit my profile data
Then my current URI route is "/profile/edit"
I see a form like the registration page
The form contains all of my user information
The password fields are blank and can be left blank
I can change any or all of the information
When I submit the form
Then I am returned to my profile page
And I see a flash message telling me that my data is updated
And I see my updated information
```

DO THESE STORIES FIRST:
- User Story 17, User Profile Show Page

### Checklist:

- [x] base tests are written
- [x] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #253

child of #274

depends on #252

### PR DESCRIPTION
**Pull request by BrennanAyers**
This PR brings in the functionality for a Default User to edit their information. This is done through the /profile page, and following through on the Edit Profile button.
The Default User will be served a form that is filled out with most of their existing information, everything except for their Password. They can change none or all of the information and it will send a PATCH request to the UsersController.
Fortunately, Rails Magic covers an empty string for Password, not changing the Password in that case.

Resolves #65

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -16,6 +16,8 @@
   get '/logout', to: 'sessions#logout' #add logout to pass navigation - User Sotry 3, User Navigation
 
   get '/profile', to: 'users#show'
+  get '/profile/edit', to: 'users#edit'
+  patch 'profile/edit', to: 'users#patch'
```
Comment (WHomer) [inline config/routes.rb]: Should this be update?

**Review by WHomer — APPROVED** [review]

File: config/routes.rb (hunk 1 above)
Comment (BrennanAyers) [reply config/routes.rb]: Done!

File: spec/features/users/show_spec.rb [hunk 2]
```diff
@@ -30,7 +30,7 @@
 
       click_on "Edit Profile"
 
-      expect(current_path).to eq("/users/#{@user.id}/edit")
+      expect(current_path).to eq("/profile/edit")
```
Comment (VinceCarollo) [inline spec/features/users/show_spec.rb:33]: Per the Story! Good change.

**Review by VinceCarollo — APPROVED** [review]
Well made good job.
